bundler/lib/dependabot/bundler/file_updater.rb

```diff
@@ -102,6 +108,13 @@ def updated_vendor_cache_files(base_directory:)
         end
       end
 
+      def binary_file?(path)
+        magic = MimeMagic.by_path(path)
```

Comment:
Unfortunately `MimeMagic` seems to think that `erb` files are binary for example, it only seems to know about `.rb` files explicitly, I think we may be better off keeping our own list in that case?

Reply:
> I think we may be better off keeping our own list

Let's do it!

I started down that path - it's ~:100: if you match filenames (e.g. `Gemfile`) then a secondary pass on extensions.

I force-pushed it into oblivion while writing the PR description as:
* I didn't want to be on the hook to maintain it
* it felt a little not-invented-here

The result was the first google hit for a library I could delegate this too - thanks for vetting the implementation! ❤️ 

--- 

I pulled `find . -type f | grep -v .git | sed -e 'sF.*/FFg' -e 's/.*\.//g' | sort | uniq -c | sort -nr | head -n 25` from this here ruby monorepo:
```
 561 rb
 278 json
 156 lock - nope, these are probably Gemfile.lock
 135 xml
  67 txt
  35 toml - nope
  32 html
  31 md
  27 gradle - nope but moot for bundler ecosystem
  23 in - nope
  22 Gemfile - nope
  21 py
  19 js
  17 gemspec - nope, these are literally everywhere
  11 gem
--- i stopped checking here
  11 csproj
   9 config
   8 pom
   8 mod
   8 go
   7 build
   6 yml
   6 props
   6 php
   5 sum
```

Reply:
https://github.com/dependabot/dependabot-core/pull/2484/commits/9a926d289dff1d6e0f35cdb0dc5b64131403c8a4